### PR TITLE
Add some convenience geometry methods

### PR DESCRIPTION
**Pull request by yck011522**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by yck011522** [comment]
PR can close #896

**Comment by tomvanmele** [comment]
test fail on ironpython due to a missing fixture it seems

**Comment by jf---** [comment]
Cool! @tomvanmele pointed a preference for `reverse` over `flip`, I think that's an important point.

**Comment by yck011522** [comment]
I'm slightly inclined to keep the terminology borrowed from Rhino / Grasshopper where

- Vector (and other non tangible things, such as list) are reversed
- Lines (and other tangible entity, such as curves, meshes and surfaces) are flipped.

File: src/compas/geometry/curves/line.py [hunk 1]
```diff
@@ -297,17 +297,19 @@ def transform(self, T):
...
-        """Construct a point at a specific location along the line.
+        """Construct a point along the line at a fractional position.
 
         Parameters
         ----------
         t : float
-            The location along the line.
+            The relative position along the line as a fraction of the length of the line.
+            0.0 corresponds to the start point and 1.0 corresponds to the end point.
+            Number outside of this range are also valid and correspond to points beyond the start and end point.
```
Comment (gonzalocasas) [inline src/compas/geometry/curves/line.py]:
```suggestion
            Numbers outside of this range are also valid and correspond to points beyond the start and end point.
```

**Review by gonzalocasas — APPROVED** [review]
lgtm

about the `flip` vs `reverse`, I don't have a strong opinion, whatever you guys think it's best

**Pull request merged**